core/src/main/java/com/netflix/conductor/core/dal/ExecutionDAOFacade.java

```diff
@@ -332,19 +333,46 @@ public void removeFromPendingWorkflow(String workflowType, String workflowId) {
      * Removes the workflow from the data store.
      *
      * @param workflowId the id of the workflow to be removed
-     * @param archiveWorkflow if true, the workflow will be archived in the {@link IndexDAO} after
-     *     removal from {@link ExecutionDAO}
+     * @param archiveWorkflow if true, the workflow and associated tasks will be archived in the
+     *     {@link IndexDAO} after removal from {@link ExecutionDAO}.
      */
     public void removeWorkflow(String workflowId, boolean archiveWorkflow) {
         WorkflowModel workflow = getWorkflowModelFromDataStore(workflowId, true);
 
+        executionDAO.removeWorkflow(workflowId);
+
+        List<TaskModel> tasks = workflow.getTasks();
         try {
             removeWorkflowIndex(workflow, archiveWorkflow);
         } catch (JsonProcessingException e) {
             throw new TransientException("Workflow can not be serialized to json", e);
         }
-
-        executionDAO.removeWorkflow(workflowId);
+        tasks.forEach(
```

Comment:
Why loop through the tasks twice? A regular for loop might be easier to read here, and couldn't you do both operations through a single loop of the tasks?

Reply:
Personally I found it more readable to have two separate loops with separate error handling. When it comes to a traditional for loop: I wanted to keep in line with the style of the file where a for each is used instead of a traditional for loop. Either way since it’s unnecessary to loop over something twice and now that the order of method has been changed, I’ll merge the loops.